test/auto-languageclient.test.ts

```diff
@@ -1,24 +1,128 @@
+
+      async function afterEachCallback() {
+        serverManager.stopListening()
+        await serverManager.stopAllServers()
+        serverManager.terminate()
+        for (const project of atom.project.getPaths()) {
+          atom.project.removePath(project)
+        }
+        await client.deactivate()
+      }
+
+      describe("getWorkspaceFolders", () => {
+        beforeEach(beforeEachCallback)
+        afterEach(afterEachCallback)
+
+        it("returns null when no server is running", async () => {
+          const workspaceFolders = await serverManager.getWorkspaceFolders()
+          expect(workspaceFolders).toBeNull()
+        })
+        it("returns null when a single file is open", async () => {
+          await atom.workspace.open(__filename)
+          const workspaceFolders = await serverManager.getWorkspaceFolders()
+          expect(workspaceFolders).toBeNull()
+        })
+        it("gives the open workspace folders", async () => {
+          const projectPath = __dirname
+          const projectPath2 = dirname(__dirname)
+
+          const workspaceFolder = projectPathToWorkspaceFolder(projectPath)
+          const workspaceFolder2 = projectPathToWorkspaceFolder(projectPath2)
+
+          // gives the open workspace folder
+          atom.project.addPath(projectPath)
+          await serverManager.startServer(projectPath)
+          expect(await serverManager.getWorkspaceFolders()).toEqual([workspaceFolder])
+
+          // doesn't give the workspace folder if it the server is not started for that project
+          atom.project.addPath(projectPath2)
+          expect(await serverManager.getWorkspaceFolders()).toEqual([workspaceFolder])
+          await serverManager.startServer(projectPath)
+          expect(await serverManager.getWorkspaceFolders()).toEqual([workspaceFolder, workspaceFolder2])
+        })
+      })
+      describe("didChangeWorkspaceFolders", () => {
+        beforeEach(beforeEachCallback)
+        afterEach(afterEachCallback)
+
+        it("gives a notification if the projects change", async () => {
+          const projectPath = __dirname
+          const projectPath2 = dirname(__dirname)
+
+          const workspaceFolder2 = projectPathToWorkspaceFolder(projectPath2)
+
+          atom.project.addPath(projectPath)
+          const server = await serverManager.startServer(projectPath)
+
+          const spy = spyOn(server.connection, "didChangeWorkspaceFolders")
+
+          atom.project.addPath(projectPath2)
+
+          expect(spy).toHaveBeenCalledOnceWith({
+            event: {
+              added: [workspaceFolder2],
+              removed: [],
+            },
+          })
+
+          atom.project.removePath(projectPath2)
+          expect(spy).toHaveBeenCalledTimes(2)
+
+          expect(spy.calls.mostRecent().args[0]).toEqual({
+            event: {
+              added: [],
+              removed: [workspaceFolder2],
+            },
+          })
+        })
+      })
+    })
+  })
```

Comment:
These tests time out for some reason.

Reply:
This commit fixes it, but not sure why I need to call it manually. 

https://github.com/atom-community/atom-languageclient/pull/153/commits/ea66c7fdea47b1e86bd65c80d60f105467598ec0